corehq/apps/ota/views.py

```diff
@@ -58,9 +59,11 @@ def get_restore_response(domain, couch_user, since=None, version='1.0',
                             status=401)
 
     project = Domain.get_by_name(domain)
+    app = Application.objects.get(app_id) if app_id else None
     restore_config = RestoreConfig(
         project=project,
         user=couch_user.to_casexml_user(),
+        app=app,
         params=RestoreParams(
```

Comment:
(agree it makes sense to not change mobile code)

that line of thinking makes sense. i do think I am ok with breaking the "params" being explicitly tied to url params if it makes the rest of the code easier to deal with. I didn't even really think those were coupled in my mental model of this code. I think of `RestoreParams` as the set of config options for a restore and how they get initialized as a small detail. also don't feel strongly about it though.